java/arcs/jvm/host/ServiceLoaderHostRegistry.kt

```diff
@@ -18,16 +20,11 @@ import kotlin.sequences.asSequence
  * A HostRegistry that discovers the available [ArcHost]s available on this platform by using
  * Java ServiceLoader and SPI.
  */
-class ServiceLoaderHostRegistry() : AnnotationBasedHostRegistry() {
+object ServiceLoaderHostRegistry : AnnotationBasedHostRegistry() {
 
     init {
-        loadAndRegisterHostsAndParticles().forEach { host -> registerHost(host) }
-    }
-
-    companion object {
-        val instance: ServiceLoaderHostRegistry by lazy(
-            LazyThreadSafetyMode.PUBLICATION
-        ) { ServiceLoaderHostRegistry() }
+        loadAndRegisterHostsAndParticles()
+            .forEach { host -> registerHost(host) }
```

Comment:
Another uber-nit:  Since `registerHost` only requires a single param which matches the param of the lambda, you could use a function reference.

```suggestion
            .forEach(::registerHost)
```